### PR TITLE
Ac test refactor

### PR DESCRIPTION
**Pull request by RubyGemTSL**

### CODE REVIEW

File: _cucumber/features/step_definitions/community.steps.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Then(/^I should see the community page title$/) do
+  expect(@site.community.community_title).to be_truthy
+end
+
+Then(/^I should see at least "([^"]*)" promoted links$/) do |links|
+  expect(@site.community.promoted_image_links(links)).to be_truthy
```
Comment (ollyjshaw) [inline _cucumber/features/step_definitions/community.steps.rb]: I wonder if the syntax can be made a bit better? Something like
```ruby
expect(@site.community.page_title_correct?)
```

File: _cucumber/features/step_definitions/community.steps.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+end
+
+Then(/^I should see at least "([^"]*)" community links$/) do |links|
+  @site.community.clear_filters
+  expect(@site.community.community_links(links)).to be_truthy
```
Comment (ollyjshaw) [inline _cucumber/features/step_definitions/community.steps.rb]: Mostly I think the be_truthy's are misleading. All the usages are based on a method giving true or false values based on some test criteria.

File: _cucumber/cucumber.rake [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+require 'fileutils'
+
+task default: :features
+
+Cucumber::Rake::Task.new(:wip) do |t|
+  t.cucumber_opts = '_cucumber -r _cucumber/features/ --profile wip'
+end
+
+task :cleanup do
+  puts '======== Deleting old reports ang logs ========='
```
Comment (ollyjshaw) [inline _cucumber/cucumber.rake]: typo

File: _cucumber/lib/pages/primary_nav_menu.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class PrimaryNav < Base
+
+  PRIMARY_NAV_BAR = '.primary-nav'
+
+  def links(link)
+    page.within(PRIMARY_NAV_BAR) do
+      case link
+        when 'Solutions'
+          page.has_link?('Solutions', href: BASE_URL+'/solutions/')
```
Comment (ollyjshaw) [inline _cucumber/lib/pages/primary_nav_menu.rb]: Not an issue with this PR, but @paulrobinson shouldn't these primary nav links be relative? Minor point though.

**Comment by paulrobinson** [comment]
this is ok to test

File: _cucumber/features/support/env.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+require 'rspec'
+require 'capybara/poltergeist'
+require 'selenium-webdriver'
+require 'require_all'
+require 'fileutils'
+require 'rubocop'
+require 'capybara-screenshot/cucumber'
+
+require_relative 'app'
+Dir["#{File.dirname(__FILE__)}/../../lib/pages/*.rb"].each { |page| load page }
```
Comment (ollyjshaw) [inline _cucumber/features/support/env.rb]: This line is causing issues on certain systems. The ordering of the imports is not guarenteed. base.rb must be loaded first

File: _cucumber/lib/pages/Base.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
+require 'rspec'
```
Comment (ollyjshaw) [inline _cucumber/lib/pages/Base.rb]: This file should be base.rb, not Base.rb

File: _cucumber/lib/pages/home_page.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class Home < Base
```
Comment (ollyjshaw) [inline _cucumber/lib/pages/home_page.rb]: You need to explicitly import base.rb (once it's renamed) eg
```ruby
require_relative 'base.rb'
```

File: _cucumber/lib/pages/primary_nav_menu.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
+class PrimaryNav < Base
```
Comment (ollyjshaw) [inline _cucumber/lib/pages/primary_nav_menu.rb]: You need to explicitly import base.rb (once it's renamed) eg
```ruby
require_relative 'base.rb'
```

File: _cucumber/lib/pages/products_page.rb [hunk 9]
```diff
@@ -0,0 +1,39 @@
+class ProductsPage < Base
```
Comment (ollyjshaw) [inline _cucumber/lib/pages/products_page.rb]: You need to explicitly import base.rb (once it's renamed) eg
```ruby
require_relative 'base.rb'
```

File: _cucumber/lib/pages/solutions_page.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
+class SolutionsPage < Base
```
Comment (ollyjshaw) [inline _cucumber/lib/pages/solutions_page.rb]: You need to explicitly import base.rb (once it's renamed) eg
```ruby
require_relative 'base.rb'
```

File: _cucumber/features/step_definitions/community.steps.rb (hunk 2 above)
Comment (RubyGemTSL) [inline _cucumber/features/step_definitions/community.steps.rb]: OK point taken. I can change it to give a better error when failing though. Something like 

raise "Community title was not found" unless @site.community.community_title.eql?(true)

**Comment by RubyGemTSL** [comment]
@paulrobinson - I'll update based on Olly's comments so hold off on the merge

**Comment by RubyGemTSL** [comment]
OK we should be good now. @ollyjshaw is looking over the changes.

File: Gemfile.lock [hunk 11]
```diff
@@ -337,18 +377,26 @@ DEPENDENCIES
...
   ruby-duration (~> 3.1)
   sass (~> 3.4, < 3.4.6)
+  selenium-webdriver (~> 2.48.0)
   signet (~> 0.6)
+  site_prism (~> 2.7)
   slim (~> 3.0)
   therubyracer
   uglifier (~> 2.0.1)
   wraith (~> 1.3.0)
+
```
Comment (ollyjshaw) [inline Gemfile.lock]: You'll need to remove these last three lines. It causes a headache with the builds. They are coming back for me too. I'll have to have a look into it when Paul's back.

File: _docker/awestruct/Dockerfile [hunk 12]
```diff
@@ -16,12 +16,16 @@ RUN wget http://prdownloads.sourceforge.net/optipng/optipng-0.7.5.tar.gz?downloa
...
+  openssl-devel freetype-devel fontconfig-devel libicu-devel sqlite-devel \
+  libpng-devel libjpeg-devel
+
 WORKDIR /tmp
 RUN yum install -y fontconfig freetype && \
-    wget https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-linux-x86_64.tar.bz2 && \
-    mkdir -p /opt/phantomjs && \
-    tar -xjvf /tmp/phantomjs-1.9.8-linux-x86_64.tar.bz2 -C /opt/phantomjs/ && \
-    cp /opt/phantomjs/phantomjs-1.9.8-linux-x86_64/bin/phantomjs /usr/bin/phantomjs
+wget https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-linux-x86_64.tar.bz2 && \
```
Comment (ollyjshaw) [inline _docker/awestruct/Dockerfile]: Can we put the indenting back. It's signifying a sub step in the RUN instruction

File: _cucumber/features/step_definitions/homepage_steps.rb [hunk 13]
```diff
@@ -0,0 +1 @@
+
```
Comment (ollyjshaw) [inline _cucumber/features/step_definitions/homepage_steps.rb]: Does this still need to be here? Can we remove if it's not required

File: _docker/awestruct/Dockerfile [hunk 14]
```diff
@@ -16,6 +16,10 @@ RUN wget http://prdownloads.sourceforge.net/optipng/optipng-0.7.5.tar.gz?downloa
 WORKDIR /tmp/optipng-0.7.5
 RUN ./configure && make && make install
 
+RUN yum -y install gcc gcc-c++ make flex bison gperf \
```
Comment (ollyjshaw) [inline _docker/awestruct/Dockerfile]: Why do we need these. They are deps for installing ruby.

File: _docker/awestruct/Dockerfile [hunk 15]
```diff
@@ -46,4 +46,4 @@ WORKDIR /home/awestruct/developer.redhat.com
 EXPOSE 4242
 
 ENTRYPOINT [ "/bin/bash", "-l", "-c" ]
-CMD [ "rake", "git_setup clean gen[docker]"]
+CMD [ "rake", "git_setup clean gen[docker]"]
```
Comment (ollyjshaw) [inline _docker/awestruct/Dockerfile]: new line

**Comment by ollyjshaw** [comment]
It's a good set of changes now. The retry logic is cool.
Happy to :+1:  when the last little comments are addressed
@paulrobinson I'm happy to merge

**Pull request merged**